app/helpers/__tests__/Intersect.spec.js

```diff
@@ -1,19 +1,569 @@
+      lon: -79.3912,
+    },
+  },
+};
+
+/**
+ *   Testing the intersect function.  Multiple cases covered.
+ */
+describe('intersectSetIntoBins', () => {
+  dayjs.extend(duration);
+
+  beforeEach(() => {
+    MockDate.set(TEST_MOMENT_MS);
+  });
+
+  afterEach(() => {
+    MockDate.reset();
+  });
+
+  /**
+   * Simplest case, empty data
+   */
+  it('intersect of empty locations produces bins with no intersections', () => {
+    let emptyLocations = [];
+    let emptyConcernLocations = [];
+    let resultBins = intersectSetIntoBins(
+      emptyLocations,
+      emptyConcernLocations,
+    );
+    expect(resultBins.every(a => a == -1)).toBe(true);
+  });
+
+  /**
+   * Exact matches in the baseLocations and the concernLocations
+   */
+  it('intersect of known location intersections at default intervals has known result', () => {
+    // 5 locations, spread over 17 days.  Note, the final location is over the 14 days that intersect covers
+    let baseLocations = [
+      ...generateBackfillLocationArray(
+        TEST_LOCATIONS.kansascity.base,
+        TEST_MOMENT_MS,
+      ),
+      ...generateBackfillLocationArray(
+        TEST_LOCATIONS.hammerfest.base,
+        //TEST_MOMENT_MS - 3 * 24 * 60 * 60 * 1000,
+        TEST_MOMENT_MS - dayjs.duration(3, 'days').asMilliseconds(),
+      ),
+      ...generateBackfillLocationArray(
+        TEST_LOCATIONS.hobart.base,
+        TEST_MOMENT_MS - dayjs.duration(7, 'days').asMilliseconds(),
+      ),
+      ...generateBackfillLocationArray(
+        TEST_LOCATIONS.munich.base,
+        TEST_MOMENT_MS - dayjs.duration(10, 'days').asMilliseconds(),
+      ),
+      ...generateBackfillLocationArray(
+        TEST_LOCATIONS.laconcordia.base,
+        TEST_MOMENT_MS - dayjs.duration(17, 'days').asMilliseconds(),
+      ),
+    ];
+    // same locations for the concern array, at the same times
+    let concernLocations = [
+      ...generateBackfillLocationArray(
+        TEST_LOCATIONS.kansascity.concern,
+        TEST_MOMENT_MS,
+      ),
+      ...generateBackfillLocationArray(
+        TEST_LOCATIONS.hammerfest.concern,
+        TEST_MOMENT_MS - dayjs.duration(3, 'days').asMilliseconds(),
+      ),
+      ...generateBackfillLocationArray(
+        TEST_LOCATIONS.hobart.concern,
+        TEST_MOMENT_MS - dayjs.duration(7, 'days').asMilliseconds(),
+      ),
+      ...generateBackfillLocationArray(
+        TEST_LOCATIONS.munich.concern,
+        TEST_MOMENT_MS - dayjs.duration(10, 'days').asMilliseconds(),
+      ),
+      ...generateBackfillLocationArray(
+        TEST_LOCATIONS.laconcordia.concern,
+        TEST_MOMENT_MS - dayjs.duration(17, 'days').asMilliseconds(),
+      ),
+    ];
+
+    // normalize and sort
+    baseLocations = normalizeAndSortLocations(baseLocations);
+    concernLocations = normalizeAndSortLocations(concernLocations);
+
+    let resultBins = intersectSetIntoBins(baseLocations, concernLocations);
+
+    let expectedBins = getEmptyLocationBins();
+    expectedBins[0] = dayjs
```

Comment:
in a lot of these tests you test that the "hits" are correct. but you don't test that the spaces in between have 0 or -1

You have a general "empty" test, which is great. but also you want to check when doing the math on real points, that it also calculates 0, -1 in the right gaps either side

Reply:
The 0's and -1 are in fact being tested... all tests are using `expect(resultBins).toEqual(expectedBins);` to do a deep compare.  That won't pass unless *all* of the bins match the expected.

Reply:
Ah, totally missed that! Great. best of both worlds highlighting specific points, then testing everything in between 👍